Two strange errors in tests TestVariable using .copy()

Setup is as described in #3777.

The two failures are the following ones:
```
____________________ TestVariable.test_index_0d_not_a_time _____________________

self = <xarray.tests.test_units.TestVariable object at 0x7f3926b2ad00>

    def test_index_0d_not_a_time(self):
        d = np.datetime64("NaT", "ns")
        x = self.cls(["x"], [d])
>       self._assertIndexedLikeNDArray(x, d)

xarray/tests/test_variable.py:206: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <xarray.tests.test_units.TestVariable object at 0x7f3926b2ad00>
variable = <xarray.Variable (x: 1)>
<Quantity(['NaT'], 'meter')>
expected_value0 = numpy.datetime64('NaT'), expected_dtype = None

    def _assertIndexedLikeNDArray(self, variable, expected_value0, expected_dtype=None):
        """Given a 1-dimensional variable, verify that the variable is indexed
        like a numpy.ndarray.
        """
        assert variable[0].shape == ()
        assert variable[0].ndim == 0
        assert variable[0].size == 1
        # test identity
>       assert variable.equals(variable.copy())
E       AssertionError: assert False
E        +  where False = <bound method Variable.equals of <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>>(<xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>)
E        +    where <bound method Variable.equals of <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>> = <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>.equals
E        +    and   <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')> = <bound method Variable.copy of <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>>()
E        +      where <bound method Variable.copy of <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>> = <xarray.Variable (x: 1)>\n<Quantity(['NaT'], 'meter')>.copy

xarray/tests/test_variable.py:151: AssertionError
_____________________ TestVariable.test_equals_all_dtypes ______________________

self = <xarray.tests.test_units.TestVariable object at 0x7f39282b4e50>

    def test_equals_all_dtypes(self):
        for v, _ in self.example_1d_objects():
            v2 = v.copy()
>           assert v.equals(v2)
E           AssertionError: assert False
E            +  where False = <bound method Variable.equals of <xarray.Variable (x: 3)>\n<Quantity(['2000-01-01T00:00:00.000000000' '2000-01-02T00:00:00.000000000'\n '2000-01-03T00:00:00.000000000'], 'meter')>>(<xarray.Variable (x: 3)>\n<Quantity(['2000-01-01T00:00:00.000000000' '2000-01-02T00:00:00.000000000'\n '2000-01-03T00:00:00.000000000'], 'meter')>)
E            +    where <bound method Variable.equals of <xarray.Variable (x: 3)>\n<Quantity(['2000-01-01T00:00:00.000000000' '2000-01-02T00:00:00.000000000'\n '2000-01-03T00:00:00.000000000'], 'meter')>> = <xarray.Variable (x: 3)>\n<Quantity(['2000-01-01T00:00:00.000000000' '2000-01-02T00:00:00.000000000'\n '2000-01-03T00:00:00.000000000'], 'meter')>.equals

xarray/tests/test_variable.py:386: AssertionError
```

I have no idea what’s wrong here.

--- a/doc/whats-new.rst
+++ b/doc/whats-new.rst
@@ -37,6 +37,8 @@ Internal Changes
 - Removed the internal ``import_seaborn`` function which handled the deprecation of
   the ``seaborn.apionly`` entry point (:issue:`3747`).
   By `Mathias Hauser <https://github.com/mathause>`_.
+- Don't test pint integration in combination with datetime objects. (:issue:`3778`, :pull:`3788`)
+  By `Justus Magin <https://github.com/keewis>`_.
 - Changed test_open_mfdataset_list_attr to only run with dask installed
   (:issue:`3777`, :pull:`3780`).
   By `Bruno Pagani <https://github.com/ArchangeGabriel>`_.

--- a/xarray/core/duck_array_ops.py
+++ b/xarray/core/duck_array_ops.py
@@ -71,14 +71,7 @@ around = _dask_or_eager_func("around")
 isclose = _dask_or_eager_func("isclose")
 
 
-if hasattr(np, "isnat") and (
-    dask_array_type == () or hasattr(dask_array_type[0], "__array_ufunc__")
-):
-    # np.isnat is available since NumPy 1.13, so __array_ufunc__ is always
-    # supported.
-    isnat = np.isnat
-else:
-    isnat = _dask_or_eager_func("isnull", eager_module=pd)
+isnat = _dask_or_eager_func("isnat")
 isnan = _dask_or_eager_func("isnan")
 zeros_like = _dask_or_eager_func("zeros_like")
 

--- a/xarray/core/duck_array_ops.py
+++ b/xarray/core/duck_array_ops.py
@@ -71,7 +71,7 @@ around = _dask_or_eager_func("around")
 isclose = _dask_or_eager_func("isclose")
 
 
-isnat = _dask_or_eager_func("isnat")
+isnat = np.isnat
 isnan = _dask_or_eager_func("isnan")
 zeros_like = _dask_or_eager_func("zeros_like")
 

--- a/xarray/core/duck_array_ops.py
+++ b/xarray/core/duck_array_ops.py
@@ -72,7 +72,7 @@ isclose = _dask_or_eager_func("isclose")
 
 
 if hasattr(np, "isnat") and (
-    dask_array is None or hasattr(dask_array_type, "__array_ufunc__")
+    dask_array_type == () or hasattr(dask_array_type[0], "__array_ufunc__")
 ):
     # np.isnat is available since NumPy 1.13, so __array_ufunc__ is always
     # supported.

--- a/xarray/tests/test_units.py
+++ b/xarray/tests/test_units.py
@@ -1333,6 +1333,7 @@ def delete_attrs(*to_delete):
     "test_index_0d_datetime",
     "test_index_0d_timedelta64",
     "test_0d_time_data",
+    "test_index_0d_not_a_time",
     "test_datetime64_conversion",
     "test_timedelta64_conversion",
     "test_pandas_period_index",
@@ -1354,6 +1355,15 @@ class TestVariable(VariableSubclassobjects):
         return xr.Variable(
             dims, unit_registry.Quantity(data, unit_registry.m), *args, **kwargs
         )
+
+    def example_1d_objects(self):
+        for data in [
+            range(3),
+            0.5 * np.arange(3),
+            0.5 * np.arange(3, dtype=np.float32),
+            np.array(["a", "b", "c"], dtype=object),
+        ]:
+            yield (self.cls("x", data), data)
 
     @pytest.mark.parametrize(
         "func",

